File: introduction/ITP1_9_B-Shuffle.cpp
#include <iostream>
using namespace std;
int main() {
  string s, t;
  int n, h;
  while(cin >> s, s != "-") {
    cin >> n;
    for(int i = 0; i < n; i++) {
      cin >> h;
      s = s.substr(h) + s.substr(0, h);
    }
    cout << s << endl;
  }
}

File: introduction/ITP1_9_B-Shuffle.php
<?php
while(($s = trim(fgets(STDIN))) != '-') {
  $n = fgets(STDIN);
  for($i = 0; $i < $n; $i++) {
    $h = trim(fgets(STDIN));
    $s = substr($s, $h).substr($s, 0, $h);
  }
  echo $s.PHP_EOL;
}
